pkg/controllers/provisioningv2/rke2/rkecontrolplane/controlplane.go

```
@@ -97,7 +98,13 @@ func (h *handler) doRemove(obj *rkev1.RKEControlPlane) func() (string, error) {
 
 		logrus.Debugf("[rkecontrolplane] (%s/%s) listed %d machines during removal", obj.Namespace, obj.Name, len(machines))
 		logrus.Tracef("[rkecontrolplane] (%s/%s) machine list: %+v", obj.Namespace, obj.Name, machines)
+		allMachines := append(machines, otherMachines...)
+		for _, machine := range allMachines {
+			if machine.DeletionTimestamp == nil {
+				_ = h.machineClient.Delete(machine.Namespace, machine.Name, &metav1.DeleteOptions{})
```

We shouldn't ignore the error.

---

fixed.